### PR TITLE
Replace `is.atomic` calls with checkmate equivalents

### PR DESCRIPTION
**Pull request by vedhav**
Closes https://github.com/insightsengineering/coredev-tasks/issues/508

`is.atomic` breaks for the `NULL` input in R devel. However, checkmate retains the past behavior, so we should replace them when `NULL` is possible as inputs.

### CODE REVIEW

File: R/choices_selected.R [hunk 1]
```diff
@@ -130,10 +130,18 @@ choices_selected <- function(choices,
...
+    checkmate::check_class(choices, "delayed_data")
+  )
+  checkmate::assert(
+    checkmate::check_atomic(selected),
+    checkmate::check_multi_class(selected, c("delayed_data", "all_choices"))
+  )
+  checkmate::assert(
+    checkmate::check_flag(keep_order),
+    checkmate::check_flag(fixed)
+  )
```
Comment (averissimo) [inline R/choices_selected.R:144]: Why the change in this? it wasn't an `OR` operation

```suggestion
  checkmate::assert_flag(keep_order)
  checkmate::assert_flag(fixed)
```

File: R/resolve.R [hunk 2]
```diff
@@ -152,7 +152,7 @@ resolve_delayed_expr <- function(x, ds, is_value_choices) {
 
   # check returned value
   if (is_value_choices) {
-    if (!is.atomic(res) || anyDuplicated(res)) {
+    if (!isTRUE(checkmate::check_atomic(res)) || anyDuplicated(res)) {
```
Comment (averissimo) [inline R/resolve.R:155]: `check_*` returns a string if `FALSE`, it doesn't matter here due to `isTRUE`, but it's a bit more accurate.

```suggestion
    if (!isTRUE(checkmate::test_atomic(res)) || anyDuplicated(res)) {
```

File: R/select_spec.R [hunk 3]
```diff
@@ -116,8 +116,16 @@ select_spec.delayed_data <- function(choices, # nolint
                                      always_selected = NULL,
                                      ordered = FALSE,
                                      label = NULL) {
-  stopifnot(is.null(selected) || is.atomic(selected) || inherits(selected, "delayed_data"))
-  stopifnot(is.null(choices) || is.atomic(choices) || inherits(choices, "delayed_data"))
+  checkmate::assert(
+    checkmate::check_null(selected),
```
Comment (averissimo) [inline R/select_spec.R:120]: Are you adding the `check_null` as a sanity check in case `checkmate changes the behaviour?

If so, then should we apply everywhere else?

``` r
checkmate::check_atomic(NULL)
#> [1] TRUE
```

**Review by averissimo — COMMENTED** [review]
Some minor comments, mostly discussion.

File: R/resolve.R (hunk 2 above)
Comment (vedhav) [reply R/resolve.R:155]: Thanks! I didn't know such a function existed.

File: R/select_spec.R (hunk 3 above)
Comment (vedhav) [reply R/select_spec.R:120]: Since we're using checkmate assert and check syntax I felt better using `check_null` rather than `is.null` here. Nothing more that that :)

File: R/choices_selected.R (hunk 1 above)
Comment (vedhav) [reply R/choices_selected.R:144]: True, I must have mixed it up while implementing this for `||` cases. Thanks for catching it!

File: R/resolve.R [hunk 4]
```diff
@@ -152,7 +152,7 @@ resolve_delayed_expr <- function(x, ds, is_value_choices) {
 
   # check returned value
   if (is_value_choices) {
-    if (!is.atomic(res) || anyDuplicated(res)) {
+    if (!isTRUE(checkmate::test_atomic(res)) || anyDuplicated(res)) {
```
Comment (averissimo) [inline R/resolve.R:155]:
```suggestion
    if (isFALSE(checkmate::test_atomic(res)) || anyDuplicated(res)) {
```

**Review by averissimo — APPROVED** [review]

File: R/resolve.R (hunk 4 above)
Comment (vedhav) [reply R/resolve.R]: `!checkmate::test_atomic(res)` should work right?

Comment (averissimo) [reply R/resolve.R]: it is :facepalm:
